src/shader/structs/lights_struct.glsl.js

```diff
@@ -44,12 +45,14 @@ export const lights_struct = /* glsl */`
 		vec4 s1 = texelFetch1D( tex, i + 1u );
 		vec4 s2 = texelFetch1D( tex, i + 2u );
 		vec4 s3 = texelFetch1D( tex, i + 3u );
+		vec4 s4 = texelFetch1D( tex, i + 4u );
 
 		Light l;
 		l.position = s0.rgb;
 		l.type = int( round( s0.a ) );
 
 		l.color = s1.rgb;
+		l.visibleSurface = s4.r;
```

Comment:
The red component of sample 4 is for other data (see [here](https://github.com/gkjohnson/three-gpu-pathtracer/blob/f427d9dae2265cbf1763be1636caf8a1ad9057ef/src/uniforms/LightsInfoUniformStruct.js#L158-L169), confusingly commented as sample 5 for s4, sorry 😅) so we if we read this sample for every light then we'll have an incorrect value for some lights. Alternatively we could pick a pixel component that isn't used by any light (like sample 6 alpha) which would may be helpful for adding support to other lights later.